metals/src/main/scala/scala/meta/internal/metals/WorkspaceFolders.scala

```diff
@@ -22,11 +23,28 @@ class WorkspaceFolders(
         case (Nil, nonScala) => (List(nonScala.head), nonScala.tail)
         case t => t
       }
-    val services = scalaProjects.map(createService(_))
-    new AtomicReference(WorkspaceFoldersServices(services, nonScalaProjects))
+    val scalaServices =
+      scalaProjects.filter(_.optDelegatePath.isEmpty).map(createService)
+    val delegating = scalaProjects.flatMap { folder =>
+      for {
+        delegate <- folder.optDelegatePath
+        service <- scalaServices.find(_.path == delegate)
+      } yield DelegatingFolderService(folder, service)
```

Comment:
Why do we need a special delegating service, could we just not create a service if it's contained in another one?

Reply:
A workplace folder service is responsible for all files that are inside that folder. This is logic used for resolving which service should handle an action. So we don't need to create a `DelegatingFolderService` but we need to have that mapping somewhere. I thought this would be a bit cleaner than having a lookup but maybe not.

Reply:
Yeah, it feels a pretty complex solution for a small use case. We could probably if we don't have a folder that starts a path, we could also as a fallback check if any of those services support that anyway instead of choosing a random service. So:
1. Check if there is a folder that contains the path.
2. If none, check if any folders declare the support for a given path via buildTargets.
3. Use fallback service.

Reply:
So it won't allow for the same kind of consistency as we have now but maybe we don't care. So e.g. if we will have a parent and lib workspace folders and parent gets closed in this solution we start a service for the lib, in your solution, lib will still use fallback. But maybe we don't care so much.

Reply:
>  parent gets closed in this solution we start a service for the lib

Wouldn't it be possible to make work also? Refresh all services when removing something for example? It's not a big deal, I would try to not add too much complexity preferably.

Reply:
Yeah, I think we can do that.